Library/Homebrew/resource.rb

```diff
@@ -87,7 +88,7 @@ def stage(target = nil, &block)
       raise ArgumentError, "target directory or block is required"
     end
 
-    verify_download_integrity(fetch)
+    verify_download_integrity(fetch) unless @download_strategy == NoDownloadStrategy
```

Comment:
Would be nice to make this a helper method or push the check into `verify_download_integrity` itself.

Reply:
Is something like this ok?
```ruby
def verify_download_integrity(fn)  # fn is a {Pathname}
     #if fn.file?    # old check
     if fn and fn.file?  # fn is nil when using Metaformula
     ...
end
```

Reply:
`if fn&.file?` would be equivalent and preferable 👍